src/components/Application.tsx

```diff
@@ -512,51 +523,63 @@ export class Application extends React.Component<Props, State> {
       isSearching: true,
       selectedFeature: null,
     })
+
     catalogService.search({
       count,
       startIndex,
       bbox: this.state.bbox,
       catalogApiKey: this.state.catalogApiKey,
       ...this.state.searchCriteria,
-    })
-      .then(searchResults => this.setState({ searchResults, searchError: null, isSearching: false }))
-      .catch(searchError => this.setState({ searchError, isSearching: false }))
+    }).then(searchResults => this.setState({
+      searchResults,
+      searchError: null,
+      isSearching: false,
+    })).catch(searchError => this.setState({
+      searchError,
+      isSearching: false,
+    }))
   }
 
   private handleSelectFeature(feature) {
     if (this.state.selectedFeature === feature) {
       return  // Nothing to do
     }
-    this.setState({
-      selectedFeature: feature || null,
-    })
+
     this.navigateTo({
       pathname: this.state.route.pathname,
-      search:   (feature && feature.properties.type === TYPE_JOB) ? `?jobId=${feature.id}` : '',
+      search: feature && feature.properties.type === TYPE_JOB ? `?jobId=${feature.id}` : '',
+      selectedFeature: feature,
     })
   }
 
   private navigateTo(loc) {
     const route = generateRoute(loc)
     history.pushState(null, null, route.href)
 
-    // Update selected feature if needed
-    let {selectedFeature} = this.state
+    // Update selected feature if needed.
+    let selectedFeature = 'selectedFeature' in loc
+      ? loc.selectedFeature
+      : this.state.selectedFeature
+
     if (route.jobIds.length) {
       selectedFeature = this.state.jobs.records.find(j => route.jobIds.includes(j.id))
     }
+
+    /* TODO: Okay?
     if (!route.jobIds.length && selectedFeature && selectedFeature.properties.type === TYPE_JOB) {
       selectedFeature = null
-    }
-    else if (route.pathname !== this.state.route.pathname && selectedFeature && selectedFeature.properties.type === TYPE_SCENE) {
+    } else if (route.pathname !== this.state.route.pathname && selectedFeature && selectedFeature.properties.type === TYPE_SCENE) {
       selectedFeature = null
     }
+    */
 
     this.setState({
       route,
       selectedFeature,
+      /* TODO: Okay?
```

Comment:
Same here... we can probably just take this out

Reply:
I left in the `TODO: Okay?` comments for now to see how some of this works out in practice.  For example, with this code commented out, your bbox is preserved after your job is created and you are looking at the job list, so that if you go back to create job, it is still there.  Likewise, the features matching the search criteria remain.

I think that is better.  Perhaps others will disagree.  Leaving the code in there but commented out will make it easy to go back if needed.